### PR TITLE
update master to main

### PR DESCRIPTION
**Pull request by makuzaverite**
Update master branch to main #2799

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/syncWithUpstream.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Sync to Upstream
```
Comment (Eomm) [inline .github/workflows/syncWithUpstream.yml:1]: I don't understand the need for this workflow, could you elaborate?

Comment (makuzaverite) [reply .github/workflows/syncWithUpstream.yml]: Sorry I was using this to sync my folks

File: docs/Routes.md [hunk 2]
```diff
@@ -333,7 +333,7 @@ fastify.get('/', { logLevel: 'warn' }, (request, reply) => {
 <a name="custom-log-serializer"></a>
 ### Custom Log Serializer
 
-In some context, you may need to log a large object but it could be a waste of resources for some routes. In this case, you can define some [`serializer`](https://github.com/pinojs/pino/blob/master/docs/api.md#bindingsserializers-object) and attach them in the right context!
+In some context, you may need to log a large object but it could be a waste of resources for some routes. In this case, you can define some [`serializer`](https://github.com/pinojs/pino/blob/main/docs/api.md#bindingsserializers-object) and attach them in the right context!
```
Comment (zekth) [inline docs/Routes.md:336]: Pino is ont using `main`

File: docs/TypeScript.md [hunk 3]
```diff
@@ -377,7 +377,7 @@ This plugin guide is for Fastify plugins written in JavaScript. The steps outlin
    export default myPlugin
    ```
 
-__Note__: [fastify-plugin](https://github.com/fastify/fastify-plugin) v2.3.0 and newer, automatically adds `.default` property and a named export to the exported plugin. Be sure to `export default` and `export const myPlugin` in your typings to provide the best developer experience. For a complete example you can check out [fastify-swagger](https://github.com/fastify/fastify-swagger/blob/master/index.d.ts).
+__Note__: [fastify-plugin](https://github.com/fastify/fastify-plugin) v2.3.0 and newer, automatically adds `.default` property and a named export to the exported plugin. Be sure to `export default` and `export const myPlugin` in your typings to provide the best developer experience. For a complete example you can check out [fastify-swagger](https://github.com/fastify/fastify-swagger/blob/main/index.d.ts).
```
Comment (zekth) [inline docs/TypeScript.md:380]: fastify-swagger is not using main

File: docs/Validation-and-Serialization.md [hunk 4]
```diff
@@ -790,7 +790,7 @@ const refToSharedSchemaDefinitions = {
 - [JSON Schema](https://json-schema.org/)
 - [Understanding JSON Schema](https://spacetelescope.github.io/understanding-json-schema/)
 - [fast-json-stringify documentation](https://github.com/fastify/fast-json-stringify)
-- [Ajv documentation](https://github.com/epoberezkin/ajv/blob/master/README.md)
+- [Ajv documentation](https://github.com/epoberezkin/ajv/blob/main/README.md)
```
Comment (zekth) [inline docs/Validation-and-Serialization.md:793]: here too

File: docs/Validation-and-Serialization.md [hunk 5]
```diff
@@ -790,7 +790,7 @@ const refToSharedSchemaDefinitions = {
...
 - [Ajv i18n](https://github.com/epoberezkin/ajv-i18n)
 - [Ajv custom errors](https://github.com/epoberezkin/ajv-errors)
-- Custom error handling with core methods with error file dumping [example](https://github.com/fastify/example/tree/master/validation-messages)
+- Custom error handling with core methods with error file dumping [example](https://github.com/fastify/example/tree/main/validation-messages)
```
Comment (zekth) [inline docs/Validation-and-Serialization.md:796]: here too

File: lib/logger.js [hunk 6]
```diff
@@ -3,7 +3,7 @@
 /**
  * Code imported from `pino-http`
  * Repo: https://github.com/pinojs/pino-http
- * License: MIT (https://raw.githubusercontent.com/pinojs/pino-http/master/LICENSE)
+ * License: MIT (https://raw.githubusercontent.com/pinojs/pino-http/main/LICENSE)
```
Comment (zekth) [inline lib/logger.js:6]: here too

File: test/inject.test.js [hunk 7]
```diff
@@ -355,7 +355,7 @@ test('inject a multipart request using form-body', t => {
     })
 })
 
-// https://github.com/hapijs/shot/blob/master/test/index.js#L836
+// https://github.com/hapijs/shot/blob/main/test/index.js#L836
```
Comment (zekth) [inline test/inject.test.js:358]: here too

**Review by zekth — CHANGES_REQUESTED** [review]
Please change the ref from `master` to `main` only when necessary

**Comment by makuzaverite** [comment]
Done @zekth

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -63,8 +63,8 @@ Welcome to the team! We are happy to have you. Before you start, please complete
...
-4. Open a pull request to [`fastify/website:master`](https://github.com/fastify/website/pulls) adding yourself to the [team.yml](https://github.com/fastify/website/blob/master/src/website/data/team.yml) file. This list is also sorted alphabetically so make sure to add your name in the proper order. Use your GitHub profile icon for the `picture:` field.
+3. Open a pull request to [`fastify/fastify:main`](https://github.com/fastify/fastify/pulls) that adds your name, username, and email to the team you have choosen in the [README.md](./README.md) and [package.json](./package.json) *(if you are part of the core team)* files. The members lists are sorted alphabetically; make sure to add your name in the proper order.
+4. Open a pull request to [`fastify/website:main`](https://github.com/fastify/website/pulls) adding yourself to the [team.yml](https://github.com/fastify/website/blob/main/src/website/data/team.yml) file. This list is also sorted alphabetically so make sure to add your name in the proper order. Use your GitHub profile icon for the `picture:` field.
```
Comment (zekth) [inline CONTRIBUTING.md:67]: website is using master

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -73,8 +73,8 @@ We are thankful to you and we are really glad to have worked with you.
 We'll be really happy to see you here again if you want to come back, but for now the person that did the onboarding must:
 1. Ask the collaborator if they want to stay or not.
 1. If the collaborator can't work with us anymore, they should:
-  1. Open a pull request to [`fastify/fastify:master`](https://github.com/fastify/fastify/pulls) and move themselves to the *Past Collaborators* section.
-  2. Open a pull request to [`fastify/website:master`](https://github.com/fastify/website/pulls) and move themselves to the *Past Collaborators* section in the [team.yml](https://github.com/fastify/website/blob/master/src/website/data/team.yml) file.
+  1. Open a pull request to [`fastify/fastify:main`](https://github.com/fastify/fastify/pulls) and move themselves to the *Past Collaborators* section.
+  2. Open a pull request to [`fastify/website:main`](https://github.com/fastify/website/pulls) and move themselves to the *Past Collaborators* section in the [team.yml](https://github.com/fastify/website/blob/main/src/website/data/team.yml) file.
```
Comment (zekth) [inline CONTRIBUTING.md:77]: website is using master

File: README.md [hunk 10]
```diff
@@ -1,5 +1,5 @@
 <div align="center">
-<img src="https://github.com/fastify/graphics/raw/master/full-logo.png" width="650" height="auto"/>
+<img src="https://github.com/fastify/graphics/raw/main/full-logo.png" width="650" height="auto"/>
```
Comment (zekth) [inline README.md:2]: graphics is using master

File: README.md [hunk 11]
```diff
@@ -191,7 +191,7 @@ matters to you.
 * <a href="./docs/Serverless.md"><code><b>Serverless</b></code></a>
 * <a href="./docs/Recommendations.md"><code><b>Recommendations</b></code></a>
 
-中文文档[地址](https://github.com/fastify/docs-chinese/blob/master/README.md)
+中文文档[地址](https://github.com/fastify/docs-chinese/blob/main/README.md)
```
Comment (zekth) [inline README.md:194]: same here

File: README.md [hunk 12]
```diff
@@ -18,7 +18,7 @@
 [![NPM version](https://img.shields.io/npm/v/fastify.svg?style=flat)](https://www.npmjs.com/package/fastify)
 [![NPM downloads](https://img.shields.io/npm/dm/fastify.svg?style=flat)](https://www.npmjs.com/package/fastify)
 [![Security Responsible
-Disclosure](https://img.shields.io/badge/Security-Responsible%20Disclosure-yellow.svg)](https://github.com/nodejs/security-wg/blob/master/processes/responsible_disclosure_template.md)
+Disclosure](https://img.shields.io/badge/Security-Responsible%20Disclosure-yellow.svg)](https://github.com/nodejs/security-wg/blob/main/processes/responsible_disclosure_template.md)
```
Comment (zekth) [inline README.md:21]: same here

File: SECURITY.md [hunk 13]
```diff
@@ -9,7 +9,7 @@ Individuals who find potential vulnerabilities in Fastify are invited to complet
 
 ### How to report a vulnerability
 
-It is of the utmost importance that you read carefully [**HOW TO REPORT A VULNERABILITY**](https://github.com/nodejs/security-wg/blob/master/processes/third_party_vuln_process.md) written by the Security Working Group of Node.js.
+It is of the utmost importance that you read carefully [**HOW TO REPORT A VULNERABILITY**](https://github.com/nodejs/security-wg/blob/main/processes/third_party_vuln_process.md) written by the Security Working Group of Node.js.
```
Comment (zekth) [inline SECURITY.md:12]: here too

File: SECURITY.md [hunk 14]
```diff
@@ -24,7 +24,7 @@ When a potential vulnerability is reported and confirmed the Fastify Core Team w
 
 ## The Fastify Core team
 
-The core team is responsible for the management of [this](https://github.com/nodejs/security-wg/blob/master/processes/third_party_vuln_process.md#handling-vulnerability-reports) process.
+The core team is responsible for the management of [this](https://github.com/nodejs/security-wg/blob/main/processes/third_party_vuln_process.md#handling-vulnerability-reports) process.
```
Comment (zekth) [inline SECURITY.md:27]: here too

File: docs/Ecosystem.md [hunk 15]
```diff
@@ -56,6 +56,7 @@ Plugins maintained by the Fastify team are listed under [Core](#core) while plug
 - [`@coobaha/typed-fastify`](https://github.com/Coobaha/typed-fastify) Strongly typed routes with a runtime validation using JSON schema generated from types.
 - [`@dnlup/fastify-doc`](https://github.com/dnlup/fastify-doc) A plugin for sampling process metrics.
 - [`@dnlup/fastify-traps`](https://github.com/dnlup/fastify-traps) A plugin to close the server gracefully on `SIGINT` and `SIGTERM` signals.
+- [`apollo-server-fastify`](https://github.com/apollographql/apollo-server/tree/main/packages/apollo-server-fastify) Run an [Apollo Server](https://github.com/apollographql/apollo-server) to serve GraphQL with Fastify.
```
Comment (zekth) [inline docs/Ecosystem.md:59]: this addition is not related to this PR

File: docs/Validation-and-Serialization.md [hunk 16]
```diff
@@ -598,7 +598,7 @@ fastify.setErrorHandler(function (error, request, reply) {
 })
 ```
 
-If you want custom error response in schema without headaches and quickly, you can take a look at [`ajv-errors`](https://github.com/epoberezkin/ajv-errors). Checkout the [example](https://github.com/fastify/example/blob/master/validation-messages/custom-errors-messages.js) usage.
+If you want custom error response in schema without headaches and quickly, you can take a look at [`ajv-errors`](https://github.com/epoberezkin/ajv-errors). Checkout the [example](https://github.com/fastify/example/blob/main/validation-messages/custom-errors-messages.js) usage.
```
Comment (zekth) [inline docs/Validation-and-Serialization.md:601]: example is using master

**Review by zekth — CHANGES_REQUESTED** [review]

File: CODE_OF_CONDUCT.md [hunk 17]
```diff
@@ -29,7 +29,7 @@ For reporting issues in spaces managed by the OpenJS Foundation, for example, re
 
 The OpenJS Foundation maintains a Code of Conduct Panel (CoCP). This is a foundation-wide team established to manage escalation when a reporter believes that a report to a member project or the CPC has not been properly handled. In order to escalate to the CoCP send an email to `coc-escalation@lists.openjsf.org`.
 
-For more information, refer to the full [Code of Conduct governance document](https://github.com/openjs-foundation/cross-project-council/blob/master/CODE_OF_CONDUCT.md).
+For more information, refer to the full [Code of Conduct governance document](https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md).
```
Comment (Eomm) [inline CODE_OF_CONDUCT.md:32]: Using `HEAD` instead of `main` will works on every github repo, whenever the main branch is called:

example:

https://github.com/openjs-foundation/cross-project-council/blob/HEAD/CODE_OF_CONDUCT.md

File: README.md [hunk 18]
```diff
@@ -8,7 +8,7 @@
 ![](https://github.com/fastify/fastify/workflows/package-manager-ci/badge.svg)
 ![](https://github.com/fastify/fastify/workflows/website/badge.svg)
 [![Known Vulnerabilities](https://snyk.io/test/github/fastify/fastify/badge.svg)](https://snyk.io/test/github/fastify/fastify)
-[![Coverage Status](https://coveralls.io/repos/github/fastify/fastify/badge.svg?branch=master)](https://coveralls.io/github/fastify/fastify?branch=master)
+[![Coverage Status](https://coveralls.io/repos/github/fastify/fastify/badge.svg?branch=HEAD)](https://coveralls.io/github/fastify/fastify?branch=HEAD)
```
Comment (Eomm) [inline README.md:11]: This image doesn't work, the `HEAD` trick seems to be broken with `coveralls.io` domain

**Review by Eomm — COMMENTED** [review]
just one nint

**Review by zekth — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

@lmammino could you please confirm this should be ok for the website integration?

**Review by Eomm — APPROVED** [review]

**Comment by lmammino** [comment]
Hey sorry for the delay. The trigger of the build of the website will work. I am checking now if the download scripts on the website repo require any change. Give me a few mins!

**Comment by lmammino** [comment]
So i have a bit of a doubt:

In this line

https://github.com/fastify/website/blob/master/src/scripts/downloadReleases.js#L118

we are assuming the current HEAD version of the project will be in: https://github.com/fastify/fastify/archive/master.zip

I am not 100% sure yet but there is a chance this needs to be changed to https://github.com/fastify/fastify/archive/main.zip

**Comment by lmammino** [comment]
Ok i just confirmed by looking at another repo and that download link needs to become:

`https://github.com/fastify/fastify/archive/refs/heads/main.zip`

I'll open a PR on the website and then we can figure out how to coordinate the release

**Review by lmammino — APPROVED** [review]

**Comment by makuzaverite** [comment]
Thanks

File: tools/website.sh [hunk 19]
```diff
@@ -2,6 +2,6 @@
 
 NODE_VER=`node -v`
 
-if [[ $NODE_VER =~ ^v6 && $TRAVIS_PULL_REQUEST == "false" && $TRAVIS_BRANCH == "master" ]]; then
+if [[ $NODE_VER =~ ^v6 && $TRAVIS_PULL_REQUEST == "false" && $TRAVIS_BRANCH == "main" ]]; then
```
Comment (delvedor) [inline tools/website.sh:5]: How can this work? We are no longer using travis 🤔

Comment (mcollina) [reply tools/website.sh:5]: cc @lmammino

Comment (makuzaverite) [reply tools/website.sh:5]: I think this should be removed since we are not relying on travis

Comment (lmammino) [reply tools/website.sh:5]: If we are not using travis, we should have the same script in github actions... double checking now...

Comment (lmammino) [reply tools/website.sh:5]: Yes, I can confirm we do the call here: https://github.com/fastify/fastify/blob/cceac26664b204e6c196e48dc156a0fd761268c2/.github/workflows/website.yml#L16

Also, this file has been already updated as part of this PR using the correct branch as a trigger for the pipeline. I think the PR is still good.

I have no idea about whether we should remove this specific file or not. Are we using it somewhere else? It is definitely not used in the new GitHub actions workflow linked above, so I wouldn't expect to see this used somewhere else...

Comment (makuzaverite) [reply tools/website.sh:5]: I think we can leave it for now

Comment (mcollina) [reply tools/website.sh:5]: Could you just drop it as part of this PR?

Comment (makuzaverite) [reply tools/website.sh:5]: @mcollina  Are you talking about removing `website.sh` completely?

Comment (mcollina) [reply tools/website.sh:5]: Yes exactly

Comment (makuzaverite) [reply tools/website.sh]: Done
